manager/controllers/app/moduleinstance.go

```diff
@@ -75,21 +90,12 @@ func (m *ModuleManager) GetCopyDestination(item modules.DataInfo, destinationInt
 	}, nil
 }
 
-// SelectModuleInstances selects the necessary read/copy/write modules for the blueprint
+// SelectModuleInstances selects the necessary read/copy/write modules for the blueprint for a given data set
+// Write path is not yet implemented
 func (m *ModuleManager) SelectModuleInstances(item modules.DataInfo) ([]modules.ModuleInstanceSpec, error) {
 	instances := make([]modules.ModuleInstanceSpec, 0)
```

Comment:
@shlomitk1 I see you have merged the two functions I had into one very large function, whose flow is hard to understand in my opinion.

Reply:
Tried to simplify the function. The idea is that the read module is selected when there is a workload, and copy is selected both when the workload requires a copy, or the user requested explicitly (ingest flow). No need for special casing.